File: 0283-move-zeroes/0283-move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    let insertPos = 0;
    let len=nums.length;
    for (let i = 0; i < len; i++) {
        if (nums[i] !== 0) {
            nums[insertPos] = nums[i];
            insertPos++;
        }
    }
    while (insertPos < len) {
        nums[insertPos] = 0;
        insertPos++;
    }
};

File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int position=0;
        int length=nums.size();
        for(int i : nums){
            if(i!=0){
                nums[position]=i;
                position++;
            }
        }
        for(int i=position;i<length;i++){
            nums[i]=0;
        }
    }
};